### PR TITLE
[groupsio] Update log message

### PR DESCRIPTION
**Pull request by animeshk08**
This commit upadates the log message
for groupsio backend in case download
archive permission is disabled

Signed-off-by: animesh <animuz111@gmail.com>

### CODE REVIEW

**Comment by animeshk08** [comment]
Hi @valeriocos 
Based on these discussions: https://github.com/chaoss/grimoirelab/issues/288#issuecomment-599253074

I have updated the error log.

**Comment by animeshk08** [comment]
Hi @valeriocos 
Based on these discussions: https://github.com/chaoss/grimoirelab/issues/288#issuecomment-599253074

I have updated the error log.

**Comment by valeriocos** [comment]
Hi @animeshk08 thank you for the PR, I'll have a look at it today!

**Comment by animeshk08** [comment]
Hi @valeriocos , I agree with you. I will modify the PR and ping you. Also to check the permission for download archive I would need to call the following API. https://groups.io/api#get-group-settings Right?

**Comment by valeriocos** [comment]
It shouldn't be needed. the info about the group permission is returned at: https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/groupsio.py#L271

Each subscription has the following shape:
```
/home/slimbook/Escritorio/sources/venv/bin/python3 /snap/pycharm-community/179/plugins/python-ce/helpers/pydev/pydevd.py --multiproc --qt-support=auto --client 127.0.0.1 --port 34653 --file /home/slimbook/Escritorio/sources/perceval/bin/perceval groupsio cloudfoundry+cf-dev -e owlbot@bitergia.com -p owl113kas --from-date 2020-02-10T19:32:53
pydev debugger: process 23000 is connecting

Connected to pydev debugger (build 193.6494.30)
[2020-03-18 11:16:22,871] - Sir Perceval is on his quest.
[2020-03-18 11:16:22,877] - Looking for messages from 'https://groups.io/g/cloudfoundry+cf-dev' since 2020-02-10 19:32:53+00:00
[2020-03-18 11:16:23,878] - Downloading mboxes from 'https://groups.io/g/cloudfoundry+cf-dev'
{
    ...
    "mod_status": "sub_modstatus_none",
    "most_recent_message": "2020-03-17T15:09:47.654088476-07:00",
    "nice_group_name": "Groups.io | cf-dev",
    "num_drafts": 0,
    "object": "member_info",
    "owner_msg_notify": "sub_ownermsg_notify_subs",
    "pending_msg_notify": "sub_notify_email",
    "pending_sub_notify": "sub_notify_email",
    "perms": { <-------------------------------
        "archives_visible": true,
        "ban_members": false,
        "calendar_visible": false,
        "can_post": true,
        "can_vote": true,
        "chat_visible": false,
        "create_hashtags": true,
        "database_visible": false,
        "delete_group": false,
        "download_archives": true, <----------------
        "download_entire_group": false,
        "download_members": false,
        "edit_archives": false,
        "files_visible": false,
        "guidelines_visible": false,
        "hashtags_visible": true,
        "invite_members": false,
        "make_moderator": false,
        "manage_calendar": false,
        "manage_chats": false,
        "manage_files": false,
        "manage_group_billing": false,
        "manage_group_payments": false,
        "manage_group_settings": false,
        "manage_hashtags": false,
        "manage_integrations": false,
        "manage_member_subscription_options": false,
        "manage_members": false,
        "manage_pending_members": false,
        "manage_pending_messages": false,
        "manage_photos": false,
        "manage_polls": false,
        "manage_subgroups": false,
        "manage_subscription": true,
        "manage_wiki": false,
        "member_directory_visible": false,
        "members_visible": false,
        "mod_page": false,
        "object": "perms",
        "open_donations_visible": false,
        "photos_visible": false,
        "public_page": true,
        "remove_members": false,
        "sub_page": true,
        "subgroups_visible": true,
        "view_activity": false,
        "view_member_directory": false,
        "wiki_visible": false
    },
    "post_status": "sub_poststatus_normal",
    ...
}
```

**Comment by animeshk08** [comment]
Hi @valeriocos. I have made the changes. I think this implementation is good. WDYT?

I will update the tests as well if this implementation suits you as well :)

File: perceval/backends/core/groupsio.py [hunk 1]
```diff
@@ -265,15 +270,15 @@ def _write_archive(r, filepath):
         with open(filepath, 'wb') as fd:
             fd.write(r.raw.read())
 
-    def __find_group_id(self):
-        """Find the id of a group given its name by iterating on the list of subscriptions"""
-
+    def __find_group_info(self):
+        """Find the id and download archive permission of a group given
```
Comment (valeriocos) [inline perceval/backends/core/groupsio.py]: Please follow the format:
```
"""Find the id ....
....
"""
group_subscriptions = self.subscriptions()
```

 Example at https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/github.py#L273

File: perceval/backends/core/groupsio.py [hunk 2]
```diff
@@ -201,7 +201,12 @@ def fetch(self, from_date=None):
         if not os.path.exists(self.dirpath):
             os.makedirs(self.dirpath)
 
-        group_id = self.__find_group_id()
+        group_id, group_download_archive = self.__find_group_info()
+
+        if not group_download_archive:
+            msg = " Enable download archive permission for the group %s" % self.group_name
```
Comment (valeriocos) [inline perceval/backends/core/groupsio.py]: I have a small dobut here, do you think it's better to report the error `Download archive permission disable for the group ...` or  suggest the solution `Enable download ..` ?

Minor comment, please remove the empty space before `Enable`

Comment (animeshk08) [reply perceval/backends/core/groupsio.py]: Your version seems better. After all its an error message. Good Catch!!

File: perceval/backends/core/groupsio.py (hunk 1 above)
Comment (animeshk08) [reply perceval/backends/core/groupsio.py]: Good Catch!

**Comment by valeriocos** [comment]
Please @animeshk08 ping me when I can review the PR, thanks

**Comment by animeshk08** [comment]
> Please @animeshk08 ping me when I can review the PR, thanks

Done! :rocket:

**Pull request merged**